NIFI-7355: remote objects support

levilentz opened the pull request:
Adds support for remote objects file, additional jars,
additional class imports, and custom return objects.

Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

_Enables X functionality; fixes bug NIFI-YYYY._

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `master`)?

- [ ] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

levilentz commented:
@MikeThomsen for some reason cant add you as a reviewer. The code runs and passes all checks on it.

MikeThomsen commented:
@levilentz Here is the issue:

```
Warning:  /home/runner/work/nifi/nifi/nifi-nar-bundles/nifi-hive-bundle/nifi-hive3-processors/src/test/java/org/apache/nifi/processors/hive/TestPutHive3Streaming.java:[677,8] [deprecation] <T>assertThat(T,Matcher<? super T>) in Assert has been deprecated
  where T is a type-variable:
    T extends Object declared in method <T>assertThat(T,Matcher<? super T>)
Warning:  src/main/java/org/apache/nifi/graph/GremlinBytecodeClientService.java:[51] (imports) AvoidStarImport: Using the '.*' form of import should be avoided - java.util.*.
Error:  Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.0:check (check-style) on project nifi-other-graph-services: You have 1 Checkstyle violation. -> [Help 1]
```

Run your code with `-Pcontrib-check` on the maven call.

levilentz committed:
NIFI-7355: remote objects support

Adds support for remote objects file, additional jars,
additional class imports, and custom return objects.
